File: from-0401-to-0500/0404-sum-of-left-leaves/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func sumOfLeftLeaves(root *TreeNode) int {
    var dfs func(node *TreeNode, isLeft bool) int
    dfs = func(node *TreeNode, isLeft bool) int {
        if isLeft && node.Left == nil && node.Right == nil {
            return node.Val
        }
        result := 0
        if node.Left != nil {
            result += dfs(node.Left, true)
        }
        if node.Right != nil {
            result += dfs(node.Right, false)
        }
        return result
    }
    return dfs(root, false)
}

File: from-0401-to-0500/0404-sum-of-left-leaves/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def dfs(node, is_left: bool)-> int:
            if is_left and node.left is None and node.right is None:
                return node.val
            
            ans = 0
            if node.left:
                ans += dfs(node.left, True)
            if node.right:
                ans += dfs(node.right, False)
                
            return ans
        
        return dfs(root, False)